File: iSeconds.Domain/MediaInfo.cs
using System;
using SQLite;

namespace iSeconds.Domain
{
	public class MediaInfo : IModel
	{
		public MediaInfo (int dayId, string path, TimeSpan timeOfDay) 
		{
			this.DayId = dayId;
			this.Path = path;
			this.TimeOfDay = timeOfDay;
		}

     	public MediaInfo()
     	{
     	}

		public string GetThumbnailPath()
		{
			if (Path.Length == 0)
				return "";

			string fileName = System.IO.Path.ChangeExtension(this.Path, ".png");
			return System.IO.Path.Combine(System.IO.Path.GetDirectoryName(Path), fileName);
		}

		#region db
		[PrimaryKey, AutoIncrement]
		public int Id { get; set; }
		public int DayId { get; set; }
		// TODO: ver o tamanho para essa string
		public string Path { get; set; }
		public TimeSpan TimeOfDay { get; set; }
		#endregion
	}
}

File: jYester/src/iSeconds/Domain/MediaInfo.java
package iSeconds.Domain;

import java.text.DateFormat;
import java.util.Date;
import java.util.TimeZone;

import com.activeandroid.Model;
import com.activeandroid.annotation.Column;
import com.activeandroid.annotation.Table;

@Table(name = "MediaInfo")
public class MediaInfo extends Model {

	public MediaInfo(long dayId, String url, Date date) {
		this.dayId = dayId;
		this.path = url;
//		this.timeOfDay = date;
	}
	
	public MediaInfo() {}

	public String getThumbnailPath() {
		if (path == null)
			return new String();
		
		return removeExtension(path) + ".png";
	}
	
	//TODO extrair isso.
	public String removeExtension(String filename) {
		int extensionIndex = filename.lastIndexOf(".");
	    if (extensionIndex == -1)
	        return filename;

	    return filename.substring(0, extensionIndex);
	}
	
	public String getTimeSpanFormatted() {
		final long TicksPerMillisecond = 10000L;
		Date date = new Date(timeOfDay / TicksPerMillisecond);
		DateFormat format= DateFormat.getTimeInstance(DateFormat.SHORT);
		format.setTimeZone(TimeZone.getTimeZone("UTC"));
		return format.format(date);
	}
	
	@Column(name = "DayId")
	public long dayId; 
	
	// TODO: ver o tamanho para essa string
	@Column(name = "Path")
	public String path;
	
	// TODO: ver como tratar issokkkk 
	@Column(name = "TimeOfDay")
	public long timeOfDay;

}
